src/Backend/Synapse/Exception/Assert.php

```diff
@@ -145,38 +146,24 @@ public static function assertSameColumns(
                 /** @var SynapseColumn $destCol */
                 $destCol = $it1->current();
                 if ($sourceCol->getColumnName() !== $destCol->getColumnName()) {
-                    throw new Exception(sprintf(
-                        'Source destination columns mismatch. "%s"->"%s"',
-                        $sourceCol->getColumnName(),
-                        $destCol->getColumnName()
-                    ));
+                    throw SynapseException::createColumnsNamesMismatch($sourceCol, $destCol);
                 }
                 $sourceDef = $sourceCol->getColumnDefinition();
                 $destDef = $destCol->getColumnDefinition();
 
                 if ($sourceDef->getType() !== $destDef->getType()) {
-                    throw new Exception(sprintf(
-                        'Source destination columns mismatch. "%s"->"%s"',
-                        $sourceDef->getSQLDefinition(),
-                        $destDef->getSQLDefinition()
-                    ));
+                    throw SynapseException::createColumnsMismatch($sourceCol, $destCol);
                 }
 
                 $isLengthEquals = $sourceDef->getLength() !== $destDef->getLength()
                     && $sourceDef->getLength() !== (string) $destDef->getDefaultLength()
                     && $destDef->getLength() !== (string) $sourceDef->getDefaultLength()
                 ;
                 if ($isLengthEquals) {
-                    throw new Exception(sprintf(
-                        'Source destination columns mismatch. "%s"->"%s"',
-                        $sourceDef->getSQLDefinition(),
-                        $destDef->getSQLDefinition()
-                    ));
+                    throw SynapseException::createColumnsMismatch($sourceCol, $destCol);
                 }
             } else {
-                throw new Exception(
-                    'Tables don\'t have same number of columns.'
-                );
+                throw SynapseException::createColumnsCountMismatch();
```

Comment:
Nebylo by tady užitečné vědět jaký mismatch to je? (tj. vypsat source a destination columns nebo jejich diff)